source/common/http/filter/ratelimit.cc

```diff
@@ -11,8 +11,9 @@
 namespace Http {
 namespace RateLimit {
 
-const Http::HeaderMapPtr Filter::TOO_MANY_REQUESTS_HEADER{new Http::HeaderMapImpl{
-    {Http::Headers::get().Status, std::to_string(enumToInt(Code::TooManyRequests))}}};
+const std::unique_ptr<const Http::HeaderMap> Filter::TOO_MANY_REQUESTS_HEADER{
+    new Http::HeaderMapImpl{
```

Comment:
We do correctly join all threads before exiting main. I'm not a huge fan of the raw pointer approach because AFAIK this will cause the heap leak detector to fail on standard settings. I don't feel super strongly about this, but we do this in other places also. Thoughts?